### PR TITLE
Add copy from subquery

### PR DESCRIPTION
**Pull request by andyfengHKU**

### CODE REVIEW

File: test/test_files/tinysnb/copy/copy_from.test [hunk 1]
```diff
@@ -0,0 +1,88 @@
+-GROUP TinySnbReadTest
+-DATASET CSV tinysnb
+
+--
+
+-CASE CopyFromSubquery
+
+-STATEMENT CREATE NODE TABLE A(name STRING, PRIMARY KEY (name));
```
Comment (acquamarin) [inline test/test_files/tinysnb/copy/copy_from.test:8]: Can you add a test where we copy from a pandas data frame and copy from a npy file?

File: src/include/planner/operator/scan/logical_scan_file.h [hunk 2]
```diff
@@ -8,22 +8,28 @@ namespace planner {
...
-    explicit LogicalScanFile(binder::BoundFileScanInfo info)
-        : LogicalOperator{LogicalOperatorType::SCAN_FILE}, info{std::move(info)} {}
+    LogicalScanFile(binder::BoundFileScanInfo info, std::shared_ptr<binder::Expression> offset)
+        : LogicalOperator{LogicalOperatorType::SCAN_FILE}, info{std::move(info)}, offset{std::move(
+                                                                                      offset)} {}
 
     inline std::string getExpressionsForPrinting() const override { return std::string(); }
 
     inline const binder::BoundFileScanInfo* getInfo() const { return &info; }
+    inline bool hasOffset() const { return offset != nullptr; }
```
Comment (acquamarin) [inline src/include/planner/operator/scan/logical_scan_file.h:18]: remove inline

File: src/include/parser/scan_source.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+struct ScanSource {
+    std::string objectName;
+    std::vector<std::string> filePaths;
+    std::unique_ptr<Statement> statement;
+
+    ScanSource() = default;
+    DELETE_COPY_AND_MOVE(ScanSource);
+
+    inline bool fromObject() const { return !objectName.empty(); }
```
Comment (acquamarin) [inline src/include/parser/scan_source.h:20]: Should we make ScanSource a hierachy  structure?
We can have BaseScanSource, and FileScanSource, statementScanSource.
Then you probably don't have to implement the fromObject(), fromFiles(), fromStatement() methods.

**Review by acquamarin — APPROVED** [review]

**Comment by ray6080** [comment]
Can you open an issue on removing full materialization and assign it to me? Thanks.

**Pull request merged**